### PR TITLE
Add script performance tests page

### PR DESCRIPTION
**Pull request by gingerabsurdity**

### CODE REVIEW

File: technical-details/waves-contracts-language-description/script-performance-tests.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Script Performance Tests
+We conducted performance tests for all aspects of our scripts. For this purpose, we developed an estimator class, that computing cost of scripts in nanoseconds after compilation phase by AST (Abstract Syntax Tree) traversal. In every test, we conducted 10 tests and calculated the average cost. The performance tests' results are the following:
```
Comment (vsuharnikov) [inline technical-details/waves-contracts-language-description/script-performance-tests.md]: Actually, the script estimator computes a script's complexity (in "complexity" units, not nanoseconds). So, performance tests is a subproject made with [JMH](http://openjdk.java.net/projects/code-tools/jmh/).

File: technical-details/waves-contracts-language-description/script-performance-tests.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+|----------------|-------------------------------|-----------------------------|
+|`appendBlock_smart`| 2\,861.589 ms/op|± 108.005 ms/op|
+|`appendBlock` | 2\,069.613 ms/op|± 449.040 ms/op|
+
+## Results
+We found the most expensive functions:
+ - base58
+ - sigVerify
+
+As a result, we define the following constraint for a script cost: a script must have a size no more than `20*cost(sigVerify) ≈ 8kB`.
```
Comment (vsuharnikov) [inline technical-details/waves-contracts-language-description/script-performance-tests.md]: We can't compare cost and size :) So, it'd better to say: 
> …: a script must be faster than`20 times of sigVerify`. Also we introduced the maximum size of a script: `8kb`.

File: technical-details/waves-contracts-language-description/script-performance-tests.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+|`appendBlock_smart`| 2\,861.589 ms/op|± 108.005 ms/op|
+|`appendBlock` | 2\,069.613 ms/op|± 449.040 ms/op|
+
+## Results
+We found the most expensive functions:
+ - base58
+ - sigVerify
+
+As a result, we define the following constraint for a script cost: a script must have a size no more than `20*cost(sigVerify) ≈ 8kB`. 
+The fixed cost for each scripted unit is equal to 400\,000 _wavelets_ (Waves coins, 100\,000\,000 wavelets = 1 Wave), i.e. if you use a scripted asset (smart asset) then you pay 400\,000 wavelets, if you also have a scripted transaction then you have to pay 2 * 400\,000 wavelets.
```
Comment (vsuharnikov) [inline technical-details/waves-contracts-language-description/script-performance-tests.md]: I'm not sure (I'm not a linguist :D ), could we distinguish `cost in wavelets` and `cost in execution time`, i.e. use different terms for them?

Comment (gingerabsurdity) [reply technical-details/waves-contracts-language-description/script-performance-tests.md]: maybe fee for the cost in wavelets?

File: technical-details/waves-contracts-language-description/script-performance-tests.md (hunk 1 above)
Comment (gingerabsurdity) [reply technical-details/waves-contracts-language-description/script-performance-tests.md]: Done

File: technical-details/waves-contracts-language-description/script-performance-tests.md (hunk 2 above)
Comment (gingerabsurdity) [reply technical-details/waves-contracts-language-description/script-performance-tests.md]: Done

File: technical-details/waves-contracts-language-description/script-performance-tests.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Script Performance Tests
-We conducted performance tests for all aspects of our scripts. For this purpose, we developed an estimator class, that computing cost of scripts in nanoseconds after compilation phase by AST (Abstract Syntax Tree) traversal. In every test, we conducted 10 tests and calculated the average cost. The performance tests' results are the following:
+We conducted performance tests for all aspects of our scripts. For this purpose, we developed an estimator subproject with [JMH](http://openjdk.java.net/projects/code-tools/jmh/), that computes a complexity of scripts after compilation phase by AST (Abstract Syntax Tree) traversal in special _complexity units_. _Complexity units_ is a measure of the script's relative cost: we found out the most expensive operation in terms of computational complexity and defined it equal to 100 complexity units. In every test, we conducted 10 tests and calculated the average cost. The performance tests' results are the following:
```
Comment (vsuharnikov) [inline technical-details/waves-contracts-language-description/script-performance-tests.md]: > estimator subproject
->
> benchmark subproject

**Pull request merged**